lib/friendly_shipping/services/ups_json/api_error_handler.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'friendly_shipping/api_error'
+
+module FriendlyShipping
+  # Handles API errors by wrapping them in an API error class ({ApiError} by default) which
+  # is then wrapped in an {ApiFailure} (along with the original API request and response).
+  # Finally, the {ApiFailure} is then wrapped in a `Dry::Monads::Failure`.
+  module Services
+    class UpsJson
+      class ApiErrorHandler
```

Comment:
I've been avoiding creating a unique `ApiErrorHandler` class for each carrier and instead using the `FriendlyShipping::ApiErrorHandler` class and passing in a custom `ApiError` instance which handles parsing out the error message.

In fact, it looks like the `UpsJson::ApiError` class already exists. Couldn't you just modify that class to pull the error from the header? Then you wouldn't need this `ApiErrorHandler` to exist.